lib/ldp.js

```diff
@@ -145,18 +145,23 @@ class LDP {
 
     const ldp = this
     debug.handlers('POST -- On parent: ' + containerPath)
-    // prepare slug
+    if (container) {
+      // Containers should not receive an extension
+      extension = ''
+    }
+    // pepare slug
     if (slug) {
-      if (this.isAuxResource(slug, extension)) throw error(403, 'POST is not allowed for auxiliary resources')
       slug = decodeURIComponent(slug)
+
+      if (container) {
+        // the name of a container cannot be a valid auxiliary resource document
+        if (this._containsInvalidSuffixes(slug + '/')) slug = slug.split('.')[0]
```

Comment:
```suggestion
        if (this._containsInvalidSuffixes(slug + '/')) slug = slug.split('.')[0]
```
 I made the split on first dot, it may be better to keep string before last dot or remove the invalid suffix

Reply:
So that results in two outputs:

If the invalid string is ```/foo/bar.acl/test.ttl```
Then removing string before last dot wouldn't work because it would still be ```/foo/bar.acl/test.ttl```
And removing the invalid suffix would be ```foo/bar/test.ttl``` which is correct, but may cause confusion for the user since it is not what they expect. It should maybe throw an error in this case for clarity?

Reply:
No this code is in POST and is proposing a new valid slug. The slug is not a path, just an item name like `foo.bar.acl`

Reply:
To make it recursive, can't it simply be changed from ```if (this._containsInvalidSuffixes)``` to ```while (this._containsInvalidSuffixes)``` ? as slug would stop updating when it is valid or when ```slug = ''``

Reply:
> No this code is in POST and is proposing a new valid slug. The slug is not a path, just an item name like `foo.bar.acl`
> 
> An other problem is if slug is `foo.meta.acl` the code need to be recursive.
> 
> An easy solution is `slug = ''` why not this is valid.
@bourgeoa This doesn't matter because none of these conditions satisfy the container requirement in the ldp.post method.

Reply:
> if (container)

This is the container requirement.

Reply:
@bourgeoa If you look at last 2 commits, I added example test with recursive check and example recursive code in ldp.post